grakn-graph/src/test/java/ai/grakn/generator/GraknGraphs.java

```diff
@@ -147,30 +147,40 @@ public GraknGraphs setOpen(boolean open) {
     private final ImmutableList<Runnable> mutators = ImmutableList.of(
             () -> {
                 TypeName typeName = typeName();
-                EntityType entityType = graph.putEntityType(typeName);
+                EntityType superType = entityType();
+                EntityType entityType = graph.putEntityType(typeName).superType(superType);
                 summaryAssign(entityType, "graph", "putEntityType", typeName);
+                summary(entityType, "superType", superType);
             },
             () -> {
                 TypeName typeName = typeName();
                 ResourceType.DataType dataType = gen(ResourceType.DataType.class);
-                ResourceType resourceType = graph.putResourceType(typeName, dataType);
+                ResourceType superType = resourceType();
+                ResourceType resourceType = graph.putResourceType(typeName, dataType).superType(superType);
                 summaryAssign(resourceType, "graph", "putResourceType", typeName, dataType);
+                summary(resourceType, "superType", superType);
             },
             () -> {
                 TypeName typeName = typeName();
                 ResourceType.DataType dataType = gen(ResourceType.DataType.class);
-                ResourceType resourceType = graph.putResourceTypeUnique(typeName, dataType);
+                ResourceType superType = resourceType();
+                ResourceType resourceType = graph.putResourceTypeUnique(typeName, dataType).superType(superType);
```

Comment:
`putResourceTypeUnique ` still exists???? I think we can kill that especially since graql does not use it.